lib/linux_admin/ssh.rb

```diff
@@ -0,0 +1,94 @@
+module LinuxAdmin
+  class SSH
+    attr_reader :ip
+    attr_reader :username
+    attr_reader :private_key
+    attr_reader :agent
+
+    def initialize(ip, username, private_key)
+      @ip = ip
+      @private_key = private_key
+      @username = username
+    end
+
+    def perform_commands(commands = [], options = {})
+      require 'net/ssh'
+      if options[:agent_enabled]
+        @agent = LinuxAdmin::SSHAgent.new(@private_key, :agent_socket => options[:agent_socket])
+        @agent.with_service do |socket|
+          if block_given?
+            execute_commands(commands, :socket => socket, :stdin => options[:stdin], &Proc.new)
+          else
+            execute_commands(commands, :socket => socket, :stdin => options[:stdin])
+          end
+        end
+      elsif block_given?
+        execute_commands(commands, :stdin => options[:stdin], &Proc.new)
+      else
+        execute_commands(commands, :stdin => options[:stdin])
+      end
+    end
+
+    private
+
+    def execute_commands(commands = [], options = {})
```

Comment:
I think maybe individual parameters would be better than an options hash here, even if they have to be optional. This is a private method so we don't have to worry about keeping a specified API and parameters help to show what can be done with the method much better than an options hash.

Also, why is `commands` optional? Why would someone call this without specifying commands to be run?